### PR TITLE
fix(associations): enable overwrite the name in unique constraint

### PR DESCRIPTION
**Pull request by RobertYoung**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Have you added new tests to prevent regressions?
- [x] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/master/CONTRIBUTING.md)?

### Description of change

Allow overwriting the unique key constraint for v4

### CODE REVIEW

File: lib/model.js [hunk 1]
```diff
@@ -3212,7 +3212,8 @@ class Model {
    */
   setDataValue(key, value) {
     const originalValue = this._previousDataValues[key];
-    if (!Utils.isPrimitive(value) || value !== originalValue) {
+
+    if (!Utils.isPrimitive(value) && value !== null || value !== originalValue) {
```
Comment (sushantdhiman) [inline lib/model.js]: Remove this change, https://github.com/sequelize/sequelize/pull/9914/

**Comment by sushantdhiman** [comment]
Backport of https://github.com/sequelize/sequelize/pull/9914/

**Comment by sushantdhiman** [comment]
Thanks @RobertYoung for this backport

**Comment by aalimovs** [comment]
that was quick! @RobertYoung @sushantdhiman thanks

**Pull request merged**